content/2017/06/27/susy3.rst

```diff
@@ -0,0 +1,872 @@
+.. _`Eric Meyer’s Reset`: https://meyerweb.com/eric/tools/css/reset/
+.. _Carl: http://oddbird.net/birds/#bird-carl
+.. _Jonny: http://oddbird.net/birds/#bird-jonny
+
+At that point,
+every layout required extensive hacks
+to work across browsers,
+and CSS frameworks quickly took over.
+But there are some things you can't do
+in a pure CSS API based on classes.
+`Natalie Downe`_ was proposing
+a more flexible approach to grids and toolkits,
+and `Chris Eppstein`_ was pushing hard
+to show how `Sass`_ and `Compass`_
+could revolutionize CSS-sharing and tool APIs.
+
+.. _Natalie Downe: http://blog.natbat.net/post/46614243624/css-systems
+.. _Chris Eppstein: http://chriseppstein.github.io/blog/
+.. _Sass: http://sass-lang.com/
+.. _Compass: http://compass-style.org/
+
+Susy came out of that mix in July, 2009.
+Nothing terribly original,
+just putting the pieces together.
+I didn't have a GitHub account at the time,
+so Carl published Susy on my behalf.
+The first draft was kinda ugly –
+but a few people took interest,
+and helped me clean it up.
+Three years later,
+it had become the inspiration
+for an entire genre of Sass grid systems –
+with several creative adaptations like `Singularity`_
+to keep us on our toes.
+
+.. _Singularity: https://github.com/at-import/Singularity
+
+
+A Whole New World
+-----------------
+
+We work on a different web now.
+Browser-hacks are mostly a thing of the past,
+and we have real control over the box-model
+(one thing IE got right the first time).
+Flexbox and the CSS Grid module are providing
+real layout tools in the browser,
+with flexibility baked in.
+
+Singularity recently `shut down development`_,
+and we considered doing the same thing.
+Grid systems are on the decline for a good reason.
+It's time to move away from these tools,
+and put our layouts back in the browser.
+Instead of closing shop, though,
+we wanted to provide one more major update
+truly designed for the edge-cases
+that remain.
+
+.. _shut down development: https://snugug.com/musings/saying-goodbye-to-singularity/
+
+Susy3 is stripped down to the core –
+a grid-math module we call ``Su``,
+and a syntax module to add API sugar over top.
+There are two primary API functions,
+``span()`` and ``gutter()``,
+and a total of 4 configuration settings.
+
+That's right, Susy3 has no mixins.
+When we wrote the mixins for Susy1 and Susy2,
+floats were (practically) the only option,
+and required a fair amount of hacking help.
+Mixins provided those shortcuts,
+at the expense of user flexibility.
+But mixins also created a black-box of CSS output,
+that made it more difficult for users
+to understand the code they were writing.
+
+But the core of Susy has always been the math,
+and the "natural-language" API.
+Accessing that math through functions only
+will allow Susy to get out of your way,
+and work with any future layout technique you need:
+floats, flexbox, tables, inline-block,
+even CSS Grids (though there's no reason for it).
```

Comment:
I assume you mean "no reason for Susy if using CSS Grids", but this also sounds a bit like "no reason to use CSS Grids".